core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -365,4 +367,111 @@ public static String extractFullServiceName(String fullMethodName) {
     }
     return fullMethodName.substring(0, index);
   }
+
+  /**
+   * Creates a new builder for a {@link MethodDescriptor}.
+   */
+  @CheckReturnValue
+  public static Builder<?, ?> newBuilder() {
+    return new Builder<Object, Object>();
+  }
+
+  /**
+   * Turns this descriptor into a builder.
+   */
+  @CheckReturnValue
+  public Builder<ReqT, RespT> toBuilder() {
```

Comment:
`Attributes.newBuilder(Attributes)` is another style. Not saying that's better, but we should try to maintain some style consistency across the code path. Although we already have inconsistencies here and there, at least we can avoid introducing yet another new style to make it more fragmented.

p.s. After I wrote the above paragraph, I did a search and I realized Protobuf uses `toBuilder()`, so it's not completely divergent. I post this comment anyway to raise the awareness of the problem, but I am fine with it as-is.